src/main/java/wedlog/address/logic/parser/VendorFilterCommandParser.java

```diff
@@ -0,0 +1,98 @@
+package wedlog.address.logic.parser;
+
+import static wedlog.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static wedlog.address.logic.Messages.MESSAGE_NO_PREFIX_FOUND;
+import static wedlog.address.logic.parser.CliSyntax.PREFIX_ADDRESS;
+import static wedlog.address.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static wedlog.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static wedlog.address.logic.parser.CliSyntax.PREFIX_PHONE;
+import static wedlog.address.logic.parser.CliSyntax.PREFIX_TAG;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import wedlog.address.logic.commands.VendorFilterCommand;
+import wedlog.address.logic.parser.exceptions.ParseException;
+import wedlog.address.model.person.Vendor;
+import wedlog.address.model.person.VendorAddressPredicate;
+import wedlog.address.model.person.VendorEmailPredicate;
+import wedlog.address.model.person.VendorNamePredicate;
+import wedlog.address.model.person.VendorPhonePredicate;
+
+/**
+ * Parses user input for VendorFilter commands.
+ */
+public class VendorFilterCommandParser implements Parser<VendorFilterCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the VendorFilterCommand
+     * and returns an VendorFilterCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public VendorFilterCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_TAG);
+        Prefix[] prefixes = {PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_TAG};
```

Comment:
Just a nit, I think this would look slightly neater if you make it into a constant instead, i.e. `private static final Prefix[] PREFIXES = ...`

Reply:
Sounds good, will change it accordingly